sdk/core/azure-core/src/datetime.cpp

```diff
@@ -29,7 +34,7 @@ DateTime DateTime::Now()
   largeInt.HighPart = fileTime.dwHighDateTime;
 
   return DateTime(Duration(largeInt.QuadPart));
-#else
+#elif defined(AZ_PLATFORM_POSIX)
```

Comment:
If you go this route everywhere we do the if/elif needs an /else #error.  Otherwise the code will compile but do nothing

Reply:
And these cases this is fine. If you don't use that functionality, you don't need a compilation error. The way these ifdefs are constructed, classes and function implementations will be missing, so you get the compilation and linker errors. But if you don't use that class, if you don't invoke that function, you are not blocked by `#error`s.